test/widgets/dialog_test.dart

```diff
@@ -0,0 +1,98 @@
+import 'package:checks/checks.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:zulip/widgets/dialog.dart';
+
+import '../model/binding.dart';
+import 'dialog_checks.dart';
+import 'test_app.dart';
+
+void main() {
+  TestZulipBinding.ensureInitialized();
+
+  late BuildContext context;
+
+  const String title = "Dialog Title";
+  const String message = "Dialog message.";
+
+  Future<void> prepare(WidgetTester tester) async {
+    addTearDown(testBinding.reset);
+
+    await tester.pumpWidget(const TestZulipApp(
+      child: Scaffold(body: Placeholder())));
+    await tester.pump();
+    context = tester.element(find.byType(Placeholder));
+  }
+
+  group('showErrorDialog', () {
+    testWidgets('show error dialog', (tester) async {
+      await prepare(tester);
+
+      showErrorDialog(context: context, title: title, message: message);
+      await tester.pump();
+      checkErrorDialog(tester, expectedTitle: title, expectedMessage: message);
+
+    }, variant: const TargetPlatformVariant({TargetPlatform.android, TargetPlatform.iOS}));
+
+    testWidgets('user closes error dialog', (tester) async {
+      await prepare(tester);
+
+      showErrorDialog(context: context, title: title, message: message);
+      await tester.pump();
+
+      Widget button = checkErrorDialog(tester, expectedTitle: title);
+      await tester.tap(find.byWidget(button));
+      await tester.pump();
+      checkNoDialog(tester);
+    });
+  });
+
+  group('showSuggestedActionDialog', () {
+    const String actionButtonText = "Action";
+
+    testWidgets('show suggested action dialog', (tester) async {
+      await prepare(tester);
+
+      showSuggestedActionDialog(context: context, title: title, message: message,
+        actionButtonText: actionButtonText, onActionButtonPress: () {});
+      await tester.pump();
+
+      checkSuggestedActionDialog(tester, expectedTitle: title, expectedMessage: message,
+        expectedActionButtonText: actionButtonText);
+    }, variant: const TargetPlatformVariant({TargetPlatform.android, TargetPlatform.iOS}));
+
+    testWidgets('user presses action button', (tester) async {
+      await prepare(tester);
+
+      late String actionText;
+      const String expectedActionText = "Action performed!";
+      void onActionButtonPress() {
+        actionText = expectedActionText;
+      }
+      showSuggestedActionDialog(context: context, title: title, message: message,
+        actionButtonText: actionButtonText, onActionButtonPress: onActionButtonPress);
+      await tester.pump();
+
+      final (Widget actionButton, _) = checkSuggestedActionDialog(tester, expectedTitle: title, expectedMessage: message,
```

Comment:
nit: omit obvious types 
```suggestion
      final (actionButton, _) = checkSuggestedActionDialog(tester, expectedTitle: title, expectedMessage: message,
```

Reply:
Do things like this also count as obvious types?
`const String title = "Dialog Title";`

Reply:
and so should var be used for 
`bool wasPressed = false;`?

Reply:
> `const String title = "Dialog Title";`

Yeah, just `const title = …`.



> and so should var be used for
> `bool wasPressed = false;`?

Ah good question. No, when the variable is going to be mutated we use the type instead of `var`.

One reason for that is that it's very common for the type a mutable variable needs to be more general than the one it's initialized with — for example it gets initialized to `null` but later mutated to some other value. Dart's type inference would use only the initializer to decide the value, so would give it the narrower type (and then a type error when later trying to store a value that doesn't fit that narrow type). By contrast for final or const variables, the type chosen by type inference is almost always the type we want.

Reply:
ok interesting, good to know, thanks